src/main/java/TaskList.java

```diff
@@ -1,37 +1,41 @@
+import java.util.ArrayList;
+
 public class TaskList {
-    private Task[] list;
-    private int listPointer = 0;
+    private ArrayList<Task> list;
```

Comment:
I like the shift from an array-based implementation to using `ArrayList`. This will likely make the code more flexible and reduce the risk of `ArrayIndexOutOfBoundsException`. Was this change made in response to a specific limitation you encountered with the array-based approach?